pkg/agent/agent.go

```diff
@@ -67,18 +69,42 @@ type proc interface {
 	KillProcess() error
 }
 
-func New(log logging.Logger, kube kubernetes.Interface, plat platform.Platform, nodeName string) (*Agent, error) {
+func New(log logging.Logger, kube kubernetes.Interface, platform platform.Platform, nodeName string) (*Agent, error) {
 	if nodeName == "" {
 		return nil, errors.New("nodeName must be provided for Agent to manage")
 	}
 	var nodeclient corev1.NodeInterface
 	if kube != nil {
 		nodeclient = kube.CoreV1().Nodes()
+		// Determine which platform to use depending on the updater interface version
+		var node, err = nodeclient.Get(nodeName, v1meta.GetOptions{})
+		if err != nil {
+			return nil, errors.New("failed to retrieve node information")
+		}
+		// Get the updater interface version from the node label
+		var platformVersion = node.Labels[marker.UpdaterInterfaceVersionKey]
+		switch platformVersion {
+		case "2.0.0":
+			platform, err = update_api.New()
+			if err != nil {
+				return nil, errors.WithMessage(err, "could not setup Update API platform for agent")
+			}
+		// If the updater interface version is not specified, default back to using Updog as the platform
+		default:
+			log.Info("unknown platform version specified, defaulting to using updog")
+			fallthrough
+		case "1.0.0":
+			platform, err = updog.New()
+			if err != nil {
+				return nil, errors.WithMessage(err, "could not setup Updog platform for agent")
+			}
+		}
```

Comment:
I'd prefer to see this in the "semver" order, see below.

And I'm generally opposed to most uses of `fallthrough`, though this is a great example of its practical use.

```suggestion
	switch platformVersion {
	default:
		// If the updater interface version is not specified, default back
		// to using Updog as the platform.
		log.Info("unknown platform version specified, defaulting to using updog")
		fallthrough
	case "1.0.0":
		platform, err = updog.New()
		if err != nil {
			return nil, errors.WithMessage(err, "could not setup Updog platform for agent")
		}
	case "2.0.0":
		platform, err = update_api.New()
		if err != nil {
			return nil, errors.WithMessage(err, "could not setup Update API platform for agent")
		}
	}
```